### PR TITLE
K8SPSMDB-527: Allow tuning timeout parameters for probes

### PR DESCRIPTION
**Pull request by defbin**
[K8SPSMDB-527](https://jira.percona.com/browse/K8SPSMDB-527) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| version-service | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/d9e82eb8d2bbcb6d4e5f48c619947022b6119a4f
image: `perconalab/percona-server-mongodb-operator:PR-782-d9e82eb8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/8715c0b4abec850e331ffffec820a9e829d3fa41
image: `perconalab/percona-server-mongodb-operator:PR-782-8715c0b4`

**Review by qJkee — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| one-pod | failed |
| storage | failed |
| upgrade-consistency | failed |
| limits | failed |
| monitoring-2-0 | failed |
| self-healing | failed |
| demand-backup | failed |
| scheduled-backup | failed |
| arbiter | failed |
| self-healing-chaos | failed |
| scaling | failed |
| operator-self-healing | failed |
| security-context | failed |
| demand-backup-sharded | failed |
| service-per-pod | failed |
| operator-self-healing-chaos | failed |
| upgrade | failed |
| liveness | passed |
| rs-shard-migration | failed |
| smart-update | failed |
| upgrade-sharded | failed |
| version-service | failed |
| users | failed |
| pitr | failed |
| data-sharded | failed |
| pitr-sharded | failed |
| non-voting | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/ca9d399cf8c464af7be4827c86a2aa1eb1c712e1
image: `perconalab/percona-server-mongodb-operator:PR-782-ca9d399c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | failed |
| liveness | passed |
| operator-self-healing | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | failed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/8715c0b4abec850e331ffffec820a9e829d3fa41
image: `perconalab/percona-server-mongodb-operator:PR-782-8715c0b4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/8715c0b4abec850e331ffffec820a9e829d3fa41
image: `perconalab/percona-server-mongodb-operator:PR-782-8715c0b4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| operator-self-healing | passed |
| demand-backup-sharded | failed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| upgrade | passed |
| non-voting | passed |
| upgrade-sharded | failed |
| pitr | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/1d7008e636bc9d9c0f6a1f736c4c057462a5eea6
image: `perconalab/percona-server-mongodb-operator:PR-782-1d7008e6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| init-deploy | passed |
| version-service | passed |
| limits | passed |
| users | passed |
| scaling | passed |
| data-sharded | passed |
| security-context | passed |
| non-voting | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/1d7008e636bc9d9c0f6a1f736c4c057462a5eea6
image: `perconalab/percona-server-mongodb-operator:PR-782-1d7008e6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/1d7008e636bc9d9c0f6a1f736c4c057462a5eea6
image: `perconalab/percona-server-mongodb-operator:PR-782-1d7008e6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/1d7008e636bc9d9c0f6a1f736c4c057462a5eea6
image: `perconalab/percona-server-mongodb-operator:PR-782-1d7008e6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| demand-backup-sharded | failed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | passed |
| upgrade-sharded | failed |
| data-sharded | passed |
| non-voting | passed |
| pitr | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/fa66ba9bd9f5c4ac9bd60d18e5bd6cec5c6da065
image: `perconalab/percona-server-mongodb-operator:PR-782-fa66ba9b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/8cd92653321be00385f97320ab5326fb71743da7
image: `perconalab/percona-server-mongodb-operator:PR-782-8cd92653`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/8cd92653321be00385f97320ab5326fb71743da7
image: `perconalab/percona-server-mongodb-operator:PR-782-8cd92653`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | failed |
| one-pod | failed |
| init-deploy | failed |
| upgrade-consistency | failed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| limits | failed |
| scaling | passed |
| demand-backup | failed |
| security-context | failed |
| scheduled-backup | failed |
| rs-shard-migration | failed |
| arbiter | failed |
| demand-backup-sharded | failed |
| service-per-pod | failed |
| liveness | passed |
| smart-update | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | failed |
| pitr | failed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/5be6afa23c6b401791a52f5f0df6900f951b0e6d
image: `perconalab/percona-server-mongodb-operator:PR-782-5be6afa2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| storage | failed |
| one-pod | failed |
| upgrade-consistency | failed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| demand-backup | failed |
| scheduled-backup | failed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | failed |
| pitr-sharded | failed |
| monitoring-2-0 | failed |
| limits | failed |
| scaling | passed |
| security-context | failed |
| rs-shard-migration | failed |
| arbiter | failed |
| service-per-pod | failed |
| liveness | passed |
| smart-update | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/5be6afa23c6b401791a52f5f0df6900f951b0e6d
image: `perconalab/percona-server-mongodb-operator:PR-782-5be6afa2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | failed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | failed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | failed |
| demand-backup-sharded | failed |
| users | passed |
| upgrade | failed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/6e5708014dbbf911a3bafba43ad8d597f07b5dd2
image: `perconalab/percona-server-mongodb-operator:PR-782-6e570801`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | failed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/12a5e21537bb24fe1d210a4fbea66642e5084419
image: `perconalab/percona-server-mongodb-operator:PR-782-12a5e215`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| operator-self-healing-chaos | passed |
| smart-update | passed |
| version-service | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade | passed |
| upgrade-sharded | failed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/12a5e21537bb24fe1d210a4fbea66642e5084419
image: `perconalab/percona-server-mongodb-operator:PR-782-12a5e215`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | failed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | failed |
| data-sharded | passed |
| pitr | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/de6a717b84829f7d672e7766080c34aab4037bdc
image: `perconalab/percona-server-mongodb-operator:PR-782-de6a717b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| demand-backup-sharded | failed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/03f9ab52a67f02062b28c7918e4d0254d87c9ee6
image: `perconalab/percona-server-mongodb-operator:PR-782-03f9ab52`

**Comment by defbin** [comment]
aa2e01960e9963e27841053bb3537edf9b696f03 incremented because new probe field is added by operator as default value

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| self-healing | passed |
| limits | passed |
| self-healing-chaos | passed |
| scaling | passed |
| operator-self-healing | passed |
| security-context | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/03f9ab52a67f02062b28c7918e4d0254d87c9ee6
image: `perconalab/percona-server-mongodb-operator:PR-782-03f9ab52`

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 1]
```diff
@@ -307,67 +318,75 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
...
-			replset.ReadinessProbe.PeriodSeconds = int32(3)
+			replset.ReadinessProbe.PeriodSeconds = 3
+		}
+		if replset.ReadinessProbe.SuccessThreshold == 0 {
+			replset.ReadinessProbe.SuccessThreshold = 1
 		}
 		if replset.ReadinessProbe.FailureThreshold == 0 {
-			replset.ReadinessProbe.FailureThreshold = int32(8)
+			replset.ReadinessProbe.FailureThreshold = 8
+			if cr.CompareVersion("1.11.0") >= 0 && replset.Name == ConfigReplSetName {
```
Comment (hors) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:387]: Can I set custom values  for readiness/liveness probes   separately for all these actions ?
spec.replsets.[]
spec.sharding.mongos
spec.sharding.configsvrReplSet

File: e2e-tests/upgrade-sharded/run [hunk 2]
```diff
@@ -9,223 +9,223 @@ test_dir=$(realpath $(dirname $0))
...
+
+	desc 'check cluster after full upgrade'
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	simple_data_check "${cluster}" "${CLUSTER_SIZE}" 1 "-mongos"
+	check_applied_images "all"
+	compare_generation "2" "statefulset" "${cluster}-rs0"
+	compare_generation "2" "statefulset" "${cluster}-cfg"
+	compare_generation "3" "deployment" "${cluster}-mongos"
```
Comment (hors) [inline e2e-tests/upgrade-sharded/run:215]: why  do we have "3" there ?

**Review by hors — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 3]
```diff
@@ -307,67 +318,75 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
...
+				replset.ReadinessProbe.FailureThreshold = 3
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:388]: can you please point me to the ticket or spec why we decided to change the default for ConfigReplSet

File: deploy/cr.yaml [hunk 4]
```diff
@@ -10,7 +10,7 @@ spec:
 #  pause: true
 #  unmanaged: false
   crVersion: 1.11.0
-  image: percona/percona-server-mongodb:4.4.8-9
+  image: percona/percona-server-mongodb:4.4.9-10
   imagePullPolicy: Always
```
Comment (hors) [inline deploy/cr.yaml:14]: please update the CR (add all new options for all needed section)

File: e2e-tests/upgrade-sharded/run [hunk 5]
```diff
@@ -9,223 +9,223 @@ test_dir=$(realpath $(dirname $0))
...
+	sleep 10
+
+	desc 'check images and generation after operator upgrade'
+	wait_for_running "${cluster}-rs0" "${CLUSTER_SIZE}" "false"
+	wait_for_running "${cluster}-cfg" "${CLUSTER_SIZE}" "false"
+	wait_cluster_consistency "${cluster}" "${CLUSTER_SIZE}"
+	check_applied_images "operator"
+	compare_generation "1" "statefulset" "${cluster}-rs0"
+	compare_generation "1" "statefulset" "${cluster}-cfg"
+	compare_generation "2" "deployment" "${cluster}-mongos"
```
Comment (delgod) [inline e2e-tests/upgrade-sharded/run:194]: generation should not be changed after that operator image upgrade

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 3 above)
Comment (defbin) [reply pkg/apis/psmdb/v1/psmdb_defaults.go:388]: TPM-40

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/3293420c783cdc550c449f33458c528f4a2be426
image: `perconalab/percona-server-mongodb-operator:PR-782-3293420c`

File: e2e-tests/version-service/conf/version-service-unreachable-rs0.yml [hunk 6]
```diff
@@ -62,6 +62,8 @@ spec:
         resources:
           requests:
             storage: 1Gi
+    readinessProbe:
+      failureThreshold: 5
```
[the remark this answers is not in the document]
Comment (defbin) [reply e2e-tests/version-service/conf/version-service-unreachable-rs0.yml:66]: 268203e

File: deploy/cr.yaml (hunk 4 above)
Comment (defbin) [reply deploy/cr.yaml:14]: d14d72e

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 1 above)
Comment (defbin) [reply pkg/apis/psmdb/v1/psmdb_defaults.go:379]: yes. added default values to cr.yaml in d14d72e.
but `LivenessProbe.SuccessThreshold` are ignored and always `1` as described in [doc](https://pkg.go.dev/k8s.io/api@v0.18.0/core/v1#Probe)

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 7]
```diff
@@ -167,72 +168,81 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
...
+		if cr.Spec.Sharding.Mongos.LivenessProbe.InitialDelaySeconds == 0 {
+			cr.Spec.Sharding.Mongos.LivenessProbe.InitialDelaySeconds = initialDelaySecondsDefault
+		}
+		if cr.Spec.Sharding.Mongos.LivenessProbe.TimeoutSeconds == 0 {
+			cr.Spec.Sharding.Mongos.LivenessProbe.TimeoutSeconds = timeoutSecondsDefault
+		}
+		if cr.Spec.Sharding.Mongos.LivenessProbe.PeriodSeconds == 0 {
+			cr.Spec.Sharding.Mongos.LivenessProbe.PeriodSeconds = periodSecondsDefault
+		}
+		if cr.Spec.Sharding.Mongos.LivenessProbe.SuccessThreshold == 0 {
```
[the remark this answers is not in the document]
Comment (defbin) [reply pkg/apis/psmdb/v1/psmdb_defaults.go]: de778a7

File: e2e-tests/upgrade-sharded/run (hunk 2 above)
Comment (defbin) [reply e2e-tests/upgrade-sharded/run:215]: currently (and previously) operator does not provide `ReadinessProbe.SuccessThreshold` field for mongos deployment (but provides for cfg and rs resources).
here operator allows customise the field and provides default value for it.

in the test, after operator upgrade, it updates `percona.com/last-config-hash` annotation that has the newly added field. btw for next release we will need to revert this changes to pass the test

File: e2e-tests/upgrade-sharded/run (hunk 5 above)
Comment (defbin) [reply e2e-tests/upgrade-sharded/run:194]: https://github.com/percona/percona-server-mongodb-operator/pull/782#discussion_r745554933

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| data-sharded | passed |
| pitr | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/de778a794a51e8d728dc2ed73348c113b3414e30
image: `perconalab/percona-server-mongodb-operator:PR-782-de778a79`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| storage | passed |
| version-service | passed |
| self-healing | passed |
| users | passed |
| self-healing-chaos | passed |
| data-sharded | passed |
| operator-self-healing | passed |
| non-voting | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/de778a794a51e8d728dc2ed73348c113b3414e30
image: `perconalab/percona-server-mongodb-operator:PR-782-de778a79`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| liveness | failed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/6234e215056567bdb742a4edc4e43b1987fe521a
image: `perconalab/percona-server-mongodb-operator:PR-782-6234e215`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| init-deploy | passed |
| demand-backup | passed |
| limits | passed |
| scheduled-backup | passed |
| scaling | passed |
| demand-backup-sharded | passed |
| security-context | passed |
| upgrade | passed |
| rs-shard-migration | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/6234e215056567bdb742a4edc4e43b1987fe521a
image: `perconalab/percona-server-mongodb-operator:PR-782-6234e215`

**Review by delgod — CHANGES_REQUESTED** [review]

File: deploy/cr.yaml [hunk 8]
```diff
@@ -90,8 +90,14 @@ spec:
 #      initialDelaySeconds: 60
 #      periodSeconds: 30
 #      successThreshold: 1
-#      timeoutSeconds: 5
+#      timeoutSeconds: 10
```
[the remark this answers is not in the document]
Comment (defbin) [reply deploy/cr.yaml:93]: in the doc and previously yes. but @egegunes increase it during some bugfix.

@egegunes can you please recall way did you increase the value before?

[the remark this answers is not in the document]
Comment (defbin) [reply deploy/cr.yaml:93]: I already asked sergey to update the doc for this value

[the remark this answers is not in the document]
Comment (defbin) [reply deploy/cr.yaml:93]: @delgod take a look at https://github.com/percona/percona-server-mongodb-operator/pull/766

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| users | passed |
| upgrade-sharded | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/782/commits/b04a275e1c37412194673e17f0b4f801e947e279
image: `perconalab/percona-server-mongodb-operator:PR-782-b04a275e`

**Review by egegunes — APPROVED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**